diff -urN gcc-13.2.0/gcc/config.gcc gcc-13.2.0-patched/gcc/config.gcc
--- gcc-13.2.0/gcc/config.gcc
+++ gcc-13.2.0-patched/gcc/config.gcc
@@ -1111,6 +1111,12 @@
       gcc_cv_initfini_array=yes
       ;;
   esac
+  ;;
+*-*-b1nix*)
+  gas=yes
+  gnu_ld=yes
+  default_use_cxa_atexit=no
+  use_gcc_stdint=wrap
   ;;
 *-*-elf|arc*-*-elf*)
   # Assume that newlib is being used and so __cxa_atexit is provided.
@@ -1867,6 +1873,10 @@
 x86_64-*-elf*)
 	tm_file="${tm_file} i386/unix.h i386/att.h elfos.h newlib-stdint.h i386/i386elf.h i386/x86-64.h"
 	;;
+x86_64-*-b1nix*)
+	tm_file="${tm_file} i386/unix.h i386/att.h elfos.h newlib-stdint.h i386/i386elf.h i386/x86-64.h b1nix.h"
+	tmake_file="${tmake_file} i386/t-i386elf t-fdpbit"
+	;;
 x86_64-*-rtems*)
 	tm_file="${tm_file} i386/unix.h i386/att.h elfos.h newlib-stdint.h i386/i386elf.h i386/x86-64.h i386/rtemself.h rtems.h"
 	;;
diff -urN gcc-13.2.0/gcc/config/b1nix.h gcc-13.2.0-patched/gcc/config/b1nix.h
--- gcc-13.2.0/gcc/config/b1nix.h
+++ gcc-13.2.0-patched/gcc/config/b1nix.h
@@ -0,0 +1,16 @@
+#ifndef GCC_B1NIX_H
+#define GCC_B1NIX_H
+
+#undef TARGET_OS_CPP_BUILTINS
+#define TARGET_OS_CPP_BUILTINS()      \
+  do {                                \
+    builtin_define_std ("b1nix");     \
+    builtin_define_std ("unix");      \
+    builtin_assert ("system=b1nix");  \
+    builtin_assert ("system=unix");   \
+  } while (0)
+
+#undef STANDARD_STARTFILE_PREFIX
+#define STANDARD_STARTFILE_PREFIX "/lib/"
+
+#undef LIB_SPEC
+#define LIB_SPEC "-lb1nix"
+
+#endif
diff -urN gcc-13.2.0/gcc/system.h gcc-13.2.0-patched/gcc/system.h
--- gcc-13.2.0/gcc/system.h
+++ gcc-13.2.0-patched/gcc/system.h
@@ -197,6 +197,33 @@
 #ifdef __cplusplus
 #ifdef INCLUDE_STRING
 # include <string>
+#endif
+#if defined (INCLUDE_ALGORITHM) || !defined (HAVE_SWAP_IN_UTILITY)
+# include <algorithm>
+#endif
+#ifdef INCLUDE_LIST
+# include <list>
+#endif
+#ifdef INCLUDE_MAP
+# include <map>
+#endif
+#ifdef INCLUDE_SET
+# include <set>
+#endif
+#ifdef INCLUDE_VECTOR
+# include <vector>
+#endif
+#ifdef INCLUDE_ARRAY
+# include <array>
+#endif
+#ifdef INCLUDE_FUNCTIONAL
+# include <functional>
+#endif
+# include <cstring>
+# include <initializer_list>
+# include <new>
+# include <utility>
+# include <type_traits>
 #endif
 #endif
 
@@ -221,33 +248,6 @@
 #if !defined (errno) && defined (HAVE_DECL_ERRNO) && !HAVE_DECL_ERRNO
 extern int errno;
 #endif
-
-#ifdef __cplusplus
-#if defined (INCLUDE_ALGORITHM) || !defined (HAVE_SWAP_IN_UTILITY)
-# include <algorithm>
-#endif
-#ifdef INCLUDE_LIST
-# include <list>
-#endif
-#ifdef INCLUDE_MAP
-# include <map>
-#endif
-#ifdef INCLUDE_SET
-# include <set>
-#endif
-#ifdef INCLUDE_VECTOR
-# include <vector>
-#endif
-#ifdef INCLUDE_ARRAY
-# include <array>
-#endif
-#ifdef INCLUDE_FUNCTIONAL
-# include <functional>
-#endif
-# include <cstring>
-# include <initializer_list>
-# include <new>
-# include <utility>
-# include <type_traits>
-#endif
diff -urN gcc-13.2.0/libgcc/config.host gcc-13.2.0-patched/libgcc/config.host
--- gcc-13.2.0/libgcc/config.host
+++ gcc-13.2.0-patched/libgcc/config.host
@@ -400,6 +400,10 @@
   esac
   ;;
 case ${host} in
+x86_64-*-b1nix*)
+	extra_parts="$extra_parts crtbegin.o crtend.o"
+	tmake_file="$tmake_file i386/t-crtstuff t-fdpbit"
+	;;
 aarch64*-*-elf | aarch64*-*-rtems*)
 	extra_parts="$extra_parts crtbegin.o crtend.o crti.o crtn.o"
 	tmake_file="${tmake_file} t-fdpbit"
