diff -ur /tmp/wget-orig/wget-1.21.4/src/http.c build/wget-src/x86_64-b1nix/wget-1.21.4/src/http.c
--- /tmp/wget-orig/wget-1.21.4/src/http.c	2023-05-11 00:18:48.000000000 +0200
+++ build/wget-src/x86_64-b1nix/wget-1.21.4/src/http.c	2026-07-15 07:03:55.062874531 +0200
@@ -2839,7 +2839,7 @@
 
                   opt.metalink_over_http = false;
                   retr_err = retrieve_url (url, urlstr, NULL, NULL,
-                                           NULL, NULL, false, iri, false);
+                                           NULL, NULL, false, iri);
                   opt.metalink_over_http = _metalink_http;
 
                   url_free (url);
diff -ur /tmp/wget-orig/wget-1.21.4/src/main.c build/wget-src/x86_64-b1nix/wget-1.21.4/src/main.c
--- /tmp/wget-orig/wget-1.21.4/src/main.c	2023-05-11 00:48:28.000000000 +0200
+++ build/wget-src/x86_64-b1nix/wget-1.21.4/src/main.c	2026-07-15 07:03:49.002856854 +0200
@@ -2168,8 +2168,9 @@
             }
           else
             {
-              retrieve_url (url_parsed, t, &filename, &redirected_URL, NULL,
-                            &dt, opt.recursive, iri, true);
+              uerr_t status = retrieve_url (url_parsed, t, &filename, &redirected_URL, NULL,
+                            &dt, opt.recursive, iri);
+              inform_exit_status (status);
             }
 
           if (opt.delete_after && filename != NULL && file_exists_p (filename, NULL))
diff -ur /tmp/wget-orig/wget-1.21.4/src/metalink.c build/wget-src/x86_64-b1nix/wget-1.21.4/src/metalink.c
--- /tmp/wget-orig/wget-1.21.4/src/metalink.c	2023-05-11 00:18:48.000000000 +0200
+++ build/wget-src/x86_64-b1nix/wget-1.21.4/src/metalink.c	2026-07-15 07:03:57.216880806 +0200
@@ -480,7 +480,7 @@
               opt.metalink_over_http = false;
               DEBUGP (("Storing to %s\n", destname));
               retr_err = retrieve_url (url, mres->url, NULL, NULL,
-                                       NULL, NULL, opt.recursive, iri, false);
+                                       NULL, NULL, opt.recursive, iri);
               opt.metalink_over_http = _metalink_http;
 
               /*
@@ -1213,7 +1213,7 @@
 
   DEBUGP (("Storing to %s\n", local_file));
   retr_err = retrieve_url (url, url_str, NULL, NULL,
-                           NULL, NULL, opt.recursive, iri, false);
+                           NULL, NULL, opt.recursive, iri);
 
   if (retr_err == RETROK)
     {
diff -ur /tmp/wget-orig/wget-1.21.4/src/recur.c build/wget-src/x86_64-b1nix/wget-1.21.4/src/recur.c
--- /tmp/wget-orig/wget-1.21.4/src/recur.c	2023-05-11 00:18:48.000000000 +0200
+++ build/wget-src/x86_64-b1nix/wget-1.21.4/src/recur.c	2026-07-15 07:03:50.848862242 +0200
@@ -328,7 +328,8 @@
             {
 
               status = retrieve_url (url_parsed, url, &file, &redirected, referer,
-                                     &dt, false, i, true);
+                                     &dt, false, i);
+              inform_exit_status (status);
 
               if (html_allowed && file && status == RETROK
                   && (dt & RETROKF) && (dt & TEXTHTML))
diff -ur /tmp/wget-orig/wget-1.21.4/src/res.c build/wget-src/x86_64-b1nix/wget-1.21.4/src/res.c
--- /tmp/wget-orig/wget-1.21.4/src/res.c	2023-05-11 00:18:48.000000000 +0200
+++ build/wget-src/x86_64-b1nix/wget-1.21.4/src/res.c	2026-07-15 07:03:52.798867931 +0200
@@ -567,7 +567,7 @@
   else
     {
       err = retrieve_url (url_parsed, robots_url, file, NULL, NULL, NULL,
-                          false, i, false);
+                          false, i);
       url_free(url_parsed);
     }
 
diff -ur /tmp/wget-orig/wget-1.21.4/src/retr.c build/wget-src/x86_64-b1nix/wget-1.21.4/src/retr.c
--- /tmp/wget-orig/wget-1.21.4/src/retr.c	2023-05-11 00:18:48.000000000 +0200
+++ build/wget-src/x86_64-b1nix/wget-1.21.4/src/retr.c	2026-07-15 07:03:47.158851468 +0200
@@ -866,7 +866,7 @@
 uerr_t
 retrieve_url (struct url * orig_parsed, const char *origurl, char **file,
               char **newloc, const char *refurl, int *dt, bool recursive,
-              struct iri *iri, bool register_status)
+              struct iri *iri)
 {
   uerr_t result;
   char *url;
@@ -1167,9 +1167,6 @@
   RESTORE_METHOD;
 
 bail:
-  if (register_status)
-    inform_exit_status (result);
-
   return result;
 }
 
@@ -1211,7 +1208,8 @@
         opt.base_href = xstrdup (url);
 
       status = retrieve_url (url_parsed, url, &url_file, NULL, NULL, &dt,
-                             false, iri, true);
+                             false, iri);
+      inform_exit_status (status);
       url_free (url_parsed);
 
       if (!url_file || (status != RETROK))
@@ -1283,10 +1281,12 @@
           opt.follow_ftp = old_follow_ftp;
         }
       else
-        status = retrieve_url (parsed_url ? parsed_url : cur_url->url,
-                               cur_url->url->url, &filename,
-                               &new_file, NULL, &dt, opt.recursive, tmpiri,
-                               true);
+        {
+          status = retrieve_url (parsed_url ? parsed_url : cur_url->url,
+                                 cur_url->url->url, &filename,
+                                 &new_file, NULL, &dt, opt.recursive, tmpiri);
+          inform_exit_status (status);
+        }
       xfree (proxy);
 
       if (parsed_url)
diff -ur /tmp/wget-orig/wget-1.21.4/src/retr.h build/wget-src/x86_64-b1nix/wget-1.21.4/src/retr.h
--- /tmp/wget-orig/wget-1.21.4/src/retr.h	2023-05-11 00:18:49.000000000 +0200
+++ build/wget-src/x86_64-b1nix/wget-1.21.4/src/retr.h	2026-07-15 07:03:44.217842868 +0200
@@ -61,7 +61,7 @@
 char *fd_read_line (int);
 
 uerr_t retrieve_url (struct url *, const char *, char **, char **,
-                     const char *, int *, bool, struct iri *, bool);
+                     const char *, int *, bool, struct iri *);
 uerr_t retrieve_from_file (const char *, bool, int *);
 
 const char *retr_rate (wgint, double);
